FactoryDesignPattern.java

```diff
@@ -0,0 +1,97 @@
+import java.util.Scanner;
+
+public class FactoryDesignPattern {
+    public static void main(String[] args) {
+        Logistics l= new Logistics();
+        try {
+            l.dealWithLogistic();
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+}
+
+class Logistics {
+    void dealWithLogistic() throws Exception{
+        Scanner sc= new Scanner(System.in);
+        String transportMode= sc.nextLine();
+
+        TransportLogistic t;
```

Comment:
Why do you need this abstract class, what if it was `Mode m` and you returned the concrete product classes i.e Ship and Truck below in the switch block?